src/octoprint/plugins/corewizard/static/js/corewizard.js

```diff
@@ -12,12 +12,24 @@ $(function () {
 
         self.required = false;
 
+        self.invalidUsername = ko.pureComputed(function () {
+            return (
+                self.username() &&
+                self.username().trim() !== "" &&
+                self.username() !== self.username().trim()
+            );
+        });
+
```

Comment:
Instead of fully duplicating the logic of `validUsername` here, this should really just be used instead, negated if needed. Especially since this is logically wrong (the opposite of `a && b && c` is `!a || !b || !c`, not `!a && !b && !c` and especially not `a && !b && !c`).

Reply:
The main reason I had created it in this (arguably convoluted way) was that I didn't want the wizard and the add user dialog to start with an error state. After implementing your suggested changes it shows an error until you start typing a name. Let me know if that's the expected behavior though.

<img width="549" alt="Bildschirmfoto 2022-09-28 um 6 38 44 PM" src="https://user-images.githubusercontent.com/115072/192839347-8b7456c0-ae3e-4723-
<img width="720" alt="Bildschirmfoto 2022-09-28 um 6 47 35 PM" src="https://user-images.githubusercontent.com/115072/192841720-30a004bf-c250-4ed9-a6ba-cbb5693e95d0.png">
b94f-cc0d953f6106.png">

Reply:
Ah... yeah, in that case I'd say go with a `validUsername` but drop the `username().trim() !== ""` requirement. Maybe we should actually just disable the Create button in that case (since it would cause an error when submitted), but not mark it as an error, that would feel more intuitive. So, `validUsername` used for error marking, `enableSubmit` (or something like that) tied to `validUsername` and `username().trim() !== ""`.

Reply:
I needed to solve this a little bit differently. If the username is empty, the `validUserName()` function would always returns false. So I'm allowing an empty username to be valid, but add another function `providedUsername()` to check against so the form can't be submitted.